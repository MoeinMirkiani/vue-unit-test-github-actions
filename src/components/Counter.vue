<template>
    <p>Current value is: {{ value }}</p>
    <button @click="value++">Increase</button>
    <button @click="value--">Decrease</button>
</template>

<script setup>
import { ref } from "vue";

const value = ref(0);
</script>
