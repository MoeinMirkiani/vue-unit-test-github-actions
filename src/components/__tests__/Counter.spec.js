import { describe, it, expect } from "vitest"
import Counter from "../Counter.vue"
import { mount } from "@vue/test-utils"

describe("Counter", () => {
    it('should render properly', () => {
        const wrapper = mount(Counter)
        expect(wrapper.html()).toMatchSnapshot()
    })
})
