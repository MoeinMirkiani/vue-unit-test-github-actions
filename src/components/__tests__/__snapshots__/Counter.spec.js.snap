// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Counter > should render properly 1`] = `
"<p>Current value is: 0</p>
<button>Increase</button>
<button>Decrease</button>"
`;
